File: PokeScan/Models/Card.swift
import Foundation

enum CardLanguage: String, Codable {
    case english, japanese
}

enum PriceSource: String, Codable {
    case tcgplayer, ebay, cardmarket, aggregated
}

struct Card: Identifiable, Codable {
    let id: UUID
    var name: String
    var setNumber: String
    var setCode: String
    var language: CardLanguage
    var marketPrice: Double?
    var priceSource: PriceSource?
    var scannedAt: Date

    // "{setCode}-{setNumber replacing '/' with '-'}" e.g. base1-025-102
    var cardSKU: String {
        "\(setCode)-\(setNumber.replacingOccurrences(of: "/", with: "-"))"
    }
}

File: android/app/src/main/java/com/pokescan/app/domain/model/Card.kt
package com.snapdex.app.domain.model

enum class CardLanguage(val raw: String) {
    ENGLISH("english"),
    JAPANESE("japanese"),
}

enum class PriceSource(val raw: String) {
    TCGPLAYER("tcgplayer"),
    EBAY("ebay"),
    CARDMARKET("cardmarket"),
    AGGREGATED("aggregated"),
}

data class Card(
    val id: String,
    val name: String,
    val setNumber: String,
    val setCode: String,
    val language: CardLanguage,
    val marketPrice: Double?,
    val priceSource: PriceSource?,
    val scannedAt: Long,
    val tcgPlayerPrice: Double? = null,
    val ebayPrice: Double? = null,
    val variant: String? = null,
    val setName: String? = null,
    val setYear: Int? = null,
    val isAuthentic: Boolean? = null,
    val priceUpdatedAt: Long? = null,
    val gradeRoiPsaGrade: Int? = null,
    val gradeRoiSellValue: Double? = null,
    val gradeRoiNetProfit: Double? = null,
) {
    val cardSKU: String get() = "$setCode-${setNumber.replace("/", "-")}"
}
